apps/src/templates/census2017/CensusInaccuracyReview.jsx

```diff
@@ -0,0 +1,400 @@
+import React, { PropTypes, Component } from 'react';
+import {Table} from 'reactabular';
+import Button from '../Button';
+
+export default class CensusInaccuracyReview extends Component {
+  static propTypes = {
+    authenticityToken: PropTypes.string,
+    reportsToReview: PropTypes.array,
+  };
+
+  state = {
+    toReview: undefined,
+    notes: '',
+    resolvedReports: [],
+  };
+
+  formatTeachesCs = (teachesCs) => {
+    return (
+      <div style={{textAlign: 'center', verticalAlign: 'middle'}}>
+        {teachesCs ? teachesCs : "Null"}
+      </div>
+    );
+  };
+
+  formatComment = (comment) => {
+    return (
+      <div>
+        {comment}
+      </div>
+    );
+  };
+
+  mapHistoricSummary = (teaches_cs) => {
+    switch (teaches_cs) {
+      case null:
+        return 'Null';
+      case undefined:
+        return 'No Data';
+      default:
+        return teaches_cs.charAt(0).toUpperCase() + teaches_cs.slice(1).toLowerCase();
+    }
+  };
+
+  formatSchool = (school) => {
+    return (
+      <div>
+        {school.name}
+        <br />
+        ({school.city}, {school.state})
+      </div>
+    );
+  };
+
+  beginReviewButton = (_value, data) => {
+    if (this.state.resolvedReports.includes(data.rowData.id)) {
+      return (
+        <div style={{textAlign: 'center', verticalAlign: 'middle'}}>
+          Review Completed
+        </div>
+      );
+    } else {
+      return (
+        <Button
+          onClick={() => {this.setState({toReview: data.rowData}); $('#report-header')[0].scrollIntoView();}}
+          size="large"
+          text="Review this School"
+        />
+      );
+    }
+  };
+
+  showMainList = () => {
+    this.setState({
+      toReview: undefined,
+      notes: ''
+    });
+    $('#report-header')[0].scrollIntoView();
+  };
+
+  render() {
+    let display;
+    if (this.state.toReview) {
+      display = this.renderReview();
+    } else {
+      display = this.renderTable();
```

Comment:
This is a huge component with 2 modes. It seems perfect to be split up into sub-components. Can we break out the review and table into 2 separate components rather than separate internal render methods here? It's just for organization but that type of composition will make the code easier to review and reason about, and is more React-y.

Reply:
I split this out into two sub components.